src/cluster.c

```diff
@@ -6164,33 +6166,27 @@ void slotToKeyReplaceEntry(dictEntry *entry) {
         /* The replaced entry was the first in the list. */
         sds key = entry->key;
         unsigned int hashslot = keyHashSlot(key, sdslen(key));
-        server.cluster->slots_to_keys[hashslot].head = entry;
+        (*db->slots_to_keys)[hashslot].head = entry;
     }
 }
 
-/* Copies the slots-keys map to the specified backup structure. */
-void slotToKeyCopyToBackup(clusterSlotsToKeysData *backup) {
-    memcpy(backup, server.cluster->slots_to_keys,
-           sizeof(server.cluster->slots_to_keys));
-}
-
-/* Overwrites the slots-keys map by copying the provided backup structure. */
-void slotToKeyRestoreBackup(clusterSlotsToKeysData *backup) {
-    memcpy(server.cluster->slots_to_keys, backup,
-           sizeof(server.cluster->slots_to_keys));
-}
-
 /* Empty the slots-keys map of Redis Cluster. */
 void slotToKeyFlush(void) {
-    memset(&server.cluster->slots_to_keys, 0,
-           sizeof(server.cluster->slots_to_keys));
+    memset(*server.db->slots_to_keys, 0,
+           sizeof(*server.db->slots_to_keys));
+}
+
+/* Empty the slots-keys map linked to tempDb being asynchronously loaded. */
+void slotToKeyTempDbFlush(redisDb *tempDb) {
+    memset(*tempDb->slots_to_keys, 0,
+           sizeof(*tempDb->slots_to_keys));
```

Comment:
i think it looks odd that we have that method.
the code is identical to the other one, and the `tempDb` argument is not necessarily temp (could be the real db)...
i suggest to have just one method doing memset, which takes a db.
if we wanna avoid modifying the few calls to slotToKeyFlush, then we need to make it a wrapper method (that take no arguments and calls the other one)